assets/react/front/tutor-front.js

```diff
@@ -209,9 +232,49 @@ jQuery(document).ready(function($) {
 				}
 			});
 		},
+		isRequiredPercentage: function() {
+			const video_data = this.video_data();
+			if (!video_data) {
+				return false;
+			}
+
+			const { strict_mode, control_video_lesson_completion } = video_data;
+			if (_tutorobject.tutor_pro_url && strict_mode && control_video_lesson_completion) {
+				return true;
+			}
+			return false;
+		},
+		enable_complete_lesson_btn: function(instance) {
+			const complete_lesson_btn = $('button[name="complete_lesson_btn"]');
+			const video_data = this.video_data();
+			const completedPercentage = (instance.currentTime / instance.duration) * 100;
```

Comment:
It can be rise `infinity` when the duration zero.
Make it a separate helper to calculate percentages by handling divide by zero exception.